### DISCUSSION TITLE
Why intercept form submission for Single Page Apps?

### DISCUSSION THREAD

**Discussion by Leon0402** [ory/kratos#2543] [Q&A] [answered]
The docs mention `To submit the form, it's recommended to intercept the submission and send the form as JSON using fetch` here https://www.ory.sh/docs/kratos/self-service#form-submission-and-payload-validation-1 for Single Page Apps. 

It then gives us a link on how this can be done, but it doesn't explain why this should be done. What is the advantage?

**Comment by aeneasr** [comment, marked as answer]
The advantage is not navigating to a new page on submit, but staying in your SPA and keeping the context available
